### DISCUSSION TITLE
why recently is "run test" not available?

### DISCUSSION THREAD

**Discussion by evbo** [rust-lang/rust-analyzer#17919] [Q&A]
I run a test using VS Code GUI "run test" link directly below `#[test]`, it fails, I make a code change then....

The `run test` link is not clickable. The only thing that works is clicking the trash icon on the terminal to kill the current test process and then reload the terminal.

Anyone else experience this? There are no logs, errors, or debug info. Just appears to silently hang...

**Comment by lnicola** [comment]
No, it works for me. Are you sure your test isn't hanging? Do you get these messages at the end?

![image](https://github.com/user-attachments/assets/50035799-b27d-4d2d-8613-d6772c5bab7d)

**Reply by evbo** [reply]
I'm running unit tests that use Polars if that makes any difference. The tests say they finish... it's really odd....

**Reply by lnicola** [reply]
Can you try in a new (or different) project?

**Reply by evbo** [reply]
I cloned this repo: https://github.com/meta-rust/rust-hello-world/tree/master

And then added a `main_tests.rs` that simply asserts true

It works great. I tried adding a bunch of dependencies but cannot seem to reproduce. Whatever it is it only affects my larger repos with lots of Polars code...

Is there a way to see log output of background processes? My hunch is there's some compilation step where the `run test` UI becomes temporarily unavailable and it's not noticeable at smaller scales.

**Reply by evbo** [reply]
I just realized if I hover over the unit test function name a second popup appears with the option to `Run Test`. Even though the original `Run Test` is unclickable, this one appears to always be clickable. So no more waiting.

Strange though why for bigger projects only one of them can be clicked.
